compiler/noirc_evaluator/src/errors.rs

```diff
@@ -111,6 +112,8 @@ pub enum InternalWarning {
 pub enum InternalBug {
     #[error("Input to brillig function is in a separate subgraph to output")]
     IndependentSubgraph { call_stack: CallStack },
+    #[error("Assertion is false")]
+    AssertFailed { call_stack: CallStack },
```

Comment:
Can we change this to something that hints that we're showing this at compile-time so that users may not be as confused why other possibly-false assertions may be skipped?

E.g.
- "This assertion is always false"
- "Code after always-false assertion is unreachable"

Reply:
Done, I have added a secondary message to explain better.